r/src/type_infer.cpp

```diff
@@ -72,7 +72,8 @@ std::shared_ptr<arrow::DataType> InferArrowTypeFromVector<INTSXP>(SEXP x) {
   } else if (Rf_inherits(x, "POSIXct")) {
     auto tzone_sexp = Rf_getAttrib(x, symbols::tzone);
     if (Rf_isNull(tzone_sexp)) {
-      return timestamp(TimeUnit::MICRO);
+      auto systzone_sexp = symbols::systzone;
```

Comment:
Try this, maybe?

```suggestion
      auto systzone_sexp = symbols::systzone();
```

Reply:
Actually, no, we probably want something like https://github.com/apache/arrow/blob/56e270fda7f5647a157acd1e428d9735d6399881/r/src/table.cpp#L244-L253

Reply:
The solution might be:
```r
  auto systzone_sexp = cpp11::package("base")["Sys.timezone"];
```

Reply:
I was wrong on both of these fronts, we _actually_ want something like `sexp cpp11::package("base")["Sys.timezone"]();`

Reply:
We want the brackets at the end if we call the function on the spot, don't we? The way I did it seems to work:
```r
      auto systzone_sexp = cpp11::package("base")["Sys.timezone"];
      return timestamp(TimeUnit::MICRO, CHAR(STRING_ELT(systzone_sexp(), 0)));
```